tools/getLBT/getLBT.ts

```diff
@@ -0,0 +1,171 @@
+                    events.push({
+                        blockNumber: event.blockNumber.toString(),
+                        originNetwork: event.args[0].toString(),
+                        originTokenAddress: event.args[1],
+                        wrappedTokenAddress: event.args[2],
+                    });
+                    logger.info(
+                        `Block number: ${event.blockNumber.toString()} - wrappedTokenAddress: ${event.args[2]}`,
+                    );
+                }
+            }
+        }
+    }
+
+    logger.info(`Collecting totalSupply of every wrapped token (${events.length} tokens)...`);
+
+    // Fetch totalSupply in parallel batches
+    await executeInBatches(
+        events,
+        async (event) => {
+            const { wrappedTokenAddress } = event;
+            const contractToken = await ethers.getContractAt('TokenWrapped', wrappedTokenAddress);
+            const totalSupply = await contractToken.totalSupply({ blockTag: blockNumber });
+            event.totalSupply = totalSupply.toString();
+        },
+        concurrencyLimit,
+        'Collecting totalSupply',
+    );
+
+    if (options?.printEvents) {
+        fs.writeFileSync(path.join(__dirname, `events.json`), JSON.stringify(events, null, 2));
+    }
+
+    const tokenAddresses = [];
+    const objectInitialize = {
+        originNetwork: [],
+        originTokenAddress: [],
+        totalSupply: [],
+    };
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const event of events) {
+        tokenAddresses.push(event.wrappedTokenAddress);
+        objectInitialize.originNetwork.push(event.originNetwork);
+        objectInitialize.originTokenAddress.push(event.originTokenAddress);
+        objectInitialize.totalSupply.push(event.totalSupply);
+    }
+
+    // get eth bridge contract
+    const weth = await contract.WETHToken();
+    const originNetwork = 0;
+    const originTokenAddress = ethers.ZeroAddress;
+    let amount;
+    let initEthBalance = 0n;
+    if (weth === ethers.ZeroAddress) {
+        const ethBalance = await ethers.provider.getBalance(agglayerBridgeAddress, blockNumber);
+        initEthBalance = await ethers.provider.getBalance(agglayerBridgeAddress, 0);
+        amount = initEthBalance - ethBalance;
+    } else {
+        const contractToken = await ethers.getContractAt('TokenWrapped', weth);
+        amount = await contractToken.totalSupply({ blockTag: blockNumber });
+        const gasTokenAddres = await contract.gasTokenAddress();
+        const gasTokenNetwork = await contract.gasTokenNetwork();
+        const contractGasToken = await ethers.getContractAt('TokenWrapped', gasTokenAddres);
+        const gasTokenAmount = await contractGasToken.totalSupply({ blockTag: blockNumber });
+        objectInitialize.originNetwork.push(gasTokenNetwork);
+        objectInitialize.originTokenAddress.push(gasTokenAddres);
+        objectInitialize.totalSupply.push(gasTokenAmount.toString());
```

Comment:
now the total supply is not encessary, we can let it as it is in the script to check it tho